File: Two_Pointers/Patterns/Pattern4/Pattern4Type2.py
lst=[0]*256
left=0
right=0
s="helloworld"
maxx=0

while(right<len(s)):
    c1=s[right]
    c2=ord(c1)
    lst[c2]+=1

    while lst[c2]>1:
        d1=s[left]
        d2=ord(d1)
        lst[d2]-=1
        left+=1

    maxx=max(maxx,right-left+1)
    right+=1
print(maxx)

File: Two_Pointers/Patterns/Pattern4/Pattern4Type2.java
public class Pattern4Type2 {
    public static void main(String[] args) {
        int[] num=new int[256];
        String s="helloworld";
        int left=0;
        int right=0;
        int maxx=0;

        while(right<s.length()){
            char c=s.charAt(right);
            num[c]++;

            while(num[c]>1){
                char d=s.charAt(left++);
                num[d]--;
            }
            maxx=Math.max(maxx,right-left+1);
            right++;
        }
        System.out.println(maxx);

    }
}
